test_runner/regress/test_on_demand_wal_download.py

```diff
@@ -0,0 +1,27 @@
+from fixtures.neon_fixtures import NeonEnv
+
+
+def test_on_demand_wal_download(neon_simple_env: NeonEnv):
+    env = neon_simple_env
+    ep = env.endpoints.create_start(
+        branch_name="main",
+        endpoint_id="primary",
+        config_lines=[
+            "max_wal_size=32MB",
+            "min_wal_size=32MB",
+            "neon.logical_replication_max_snap_files=10000",
```

Comment:
> is neon.logical_replication_max_snap_files=10000 important here?
I want exclude situation when slot is dropped because too much snapshots are created while table is populated.
I don't remember now if I really encountered this problem during testing, but most likely - yes, otherwise I haven't added this line:)

> Do the min/max_wal_size settings make any difference?
As far as I understand `max_wal_size` specifies threshold for triggering checkpoint and `min_wal_size` - how much WAL segments we want to preserve. But in this test setting bother of them is needed because I want to use most aggressive WAL truncation policy, `max_wal_size` should be not  smaller than `min_wal_size` and default sizeof `min_wal_size` is 5 segments.